### DISCUSSION TITLE
Summary of all steps passed or failed (as dictionary)

### DISCUSSION THREAD

**Discussion by adityakumar1990** [behave/behave#1001] [Q&A] [answered]
Suppose we ran `behave` and there are 2 features, 2 scenarios, and 20 steps;
out of which 12 passed and 8 failed.

Now I want to save this into a dictionary-like `{"passed":12,"failed":8}`.
Any provision?

**Comment by jenisys** [comment, marked as answer]
Formatter(s) are variation-point/extension-point in `behave`.
If you need a specific output format that is not covered by the existing formatter(s) (or: reporter(s)):

* provide it in your own Formatter class
* Use it during the test run with `behave -f my_module:MyFormatter -o my_outfile.json …`
* Define a alias for the formatter/formatter-class in the „behave.ini“ config-file an use this alias on the command line

HINTS
* You can provide multiple formatters for `behave` (either on command-line and/or in „behave.ini“ config-file)
* All but the primary formatter formatter needs to specify a corresponding output file; the primary formatter uses stdout/console output by default

SEE ALSO:

* [features/formatter.user_defined.feature](https://github.com/behave/behave/blob/master/features/formatter.user_defined.feature)
*  [behave.ini](https://github.com/behave/behave/blob/master/behave.ini#L35) config-file
